src/services_models/reserves.jl

```diff
@@ -6,56 +6,59 @@ This function add the variables for reserves to the model
+                lb_value = d -> 0 )
     return
 end
 
 ################################## Reserve Requirement Constraint ##########################
 # This function can be generalized later for any constraint of type Sum(req_var) >= requirement,
 # it will only need to be specific to the names and get forecast string.
 function service_requirement_constraint!(psi_container::PSIContainer,
-                                         service::SR) where {SR<:PSY.Reserve}
-    time_steps = model_time_steps(psi_container)
+                                service::SR,
+                                model::ServiceModel{SR, RangeReserve}) where SR<:PSY.Reserve
     parameters = model_has_parameters(psi_container)
-    forecast = model_uses_forecasts(psi_container)
+    use_forecast_data = model_uses_forecasts(psi_container)
     initial_time = model_initial_time(psi_container)
-    reserve_variable = get_variable(psi_container, Symbol("$(PSY.get_name(service))_$SR"))
-    constraint_name = Symbol(PSY.get_name(service), "_requirement_$SR")
-    constraint = add_cons_container!(psi_container, constraint_name, time_steps)
-    requirement = PSY.get_requirement(service)
-    if forecast
+    time_steps = model_time_steps(psi_container)
+    constraint_name = Symbol("requirement_$SR")
+    name = PSY.get_name(service)
+    constraint = get_constraints(psi_container, constraint_name)
+    reserve_variable = get_variable(psi_container, Symbol("$(name)_$SR"))
+
+    if use_forecast_data
         ts_vector = TS.values(PSY.get_data(PSY.get_forecast(PSY.Deterministic,
                                                             service,
                                                             initial_time,
                                                             "get_requirement")))
     else
         ts_vector = ones(time_steps[end])
     end
+
     if parameters
-        param = include_parameters(psi_container, ts_vector,
-                                   UpdateRef{SR}("get_requirement"), time_steps)
+        param = get_parameters(psi_container, UpdateRef{SR}("get_requirement"))
         for t in time_steps
-            constraint[t] = JuMP.@constraint(psi_container.JuMPmodel,
-                                         sum(reserve_variable[:,t]) >= param[t]*requirement)
+            param[name, t] = PJ.add_parameter(psi_container.JuMPmodel, 
+                                                   ts_vector[t])
+            constraint[name, t] = JuMP.@constraint(psi_container.JuMPmodel,
+                                        sum(reserve_variable[:,t]) >= param[name, t])
         end
     else
         for t in time_steps
-            constraint[t] = JuMP.@constraint(psi_container.JuMPmodel,
-                                    sum(reserve_variable[:,t]) >= ts_vector[t]*requirement)
+            constraint[name, t] = JuMP.@constraint(psi_container.JuMPmodel,
+                                    sum(reserve_variable[:,t]) >= ts_vector[t])
```

Comment:
Here this change will remove the assumption that the time series is 1-0 as scaling factors. This would be inconsistent with other implementations.